std/fs.zig

```diff
@@ -877,9 +877,6 @@ pub fn readLinkC(pathname: [*]const u8, buffer: *[os.PATH_MAX]u8) ![]u8 {
 pub const OpenSelfExeError = os.OpenError || os.windows.CreateFileError || SelfExePathError;
 
 pub fn openSelfExe() OpenSelfExeError!File {
-    if (os.linux.is_the_target) {
-        return File.openReadC(c"/proc/self/exe");
```

Comment:
There is one thing that this prevents from working - if the exe is running, but deleted from disk, `readlink` gives `/path/to/exe (deleted)`, however by opening `/proc/self/exe` directly, the kernel can give the fd even though the file is deleted from disk. https://stackoverflow.com/questions/28953307/

What's happening with qemu here? It's giving you the qemu binary fd instead of the zig binary fd?

Reply:
> What's happening with qemu here? It's giving you the qemu binary fd instead of the zig binary fd?

Yup, this happens because I use `binfmt_misc` to launch the ELF file.

> There is one thing that this prevents from working 

Bleh, let's use `AT_EXECFN` + `readlink` then?

Reply:
> Yup, this happens because I use `binfmt_misc` to launch the ELF file.

Here are 2 other options:

```
./zig build test-std -Dskip-release  -Denable-wine -Denable-qemu -Denable-foreign-glibc=/home/andy/Downloads/glibc/multi/install/glibcs
```

(you probably don't have glibc builds since that took literally an entire day and 14G on my hard drive, just delete that parameter. also wine not needed for this case. we could also add a -Dtarget=riscv64-linux option to build.zig)

```
./zig test ../std/std.zig -target riscv64-linux --test-cmd qemu-riscv64 --test-cmd-bin --override-std-dir ../std
```

> Bleh, let's use `AT_EXECFN` + `readlink` then?

Meaning get it from the auxval, if available, and fallback to readlink? And you mean `AT_EXECFD` right? It looks like `AT_EXECFN` would give the same thing as readlink /proc/self/exe.

Here's my proposal:

 * Getting info from the aux val is the best way to get the information, because it requires no syscall, and no dependency on procfs. `openSelfExe` should prefer `AT_EXECFD` and `selfExePath` should prefer `AT_EXECFN`. If we're building an exe (std/special/start.zig included and calls main) then we have the aux val available. If we're linking libc, then we have `getauxval` available. The only time this is not available is if building a library and not linking libc. But we should know at comptime whether the auxval is available.

 * Next, we fall back to `open("/proc/self/exe")` (status quo). This should avoid your binfmt_misc qemu issue because the first bullet point will be the one that is triggered for std lib tests.

One issue to solve is if we call open() to get the self exe fd, that's an open resource. If we get it from `AT_EXECFD`, then it doesn't count as an open resource. We could use dup to make it look the same, but that introduces an unnecessary possible failure.

My idea to solve this would be yet another function called `openSelfExeAuxval` which asserts that the auxval is available, asserts the target is linux, guarantees success, and the returned file must not be closed. `openSelfExe` would remain higher level, cross platform. On Linux it would call `openSelfExeAuxval` if available, and then `dup`.